### PR TITLE
JNG-5844 Allow overwrite with create on Single Association Relation

### PR DESCRIPTION
**Pull request by gaborflorian**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-5844" title="JNG-5844" target="_blank">BugJNG-5844</a>  Allow overwrite with create on Single Association Relation
  </td></table>
  <br />
 

JNG-5844 Allow overwrite with create on Single Association Relation

### CODE REVIEW

File: judo-runtime-core-dao-rdbms/src/main/java/hu/blackbelt/judo/runtime/core/dao/rdbms/RdbmsDAOImpl.java [hunk 1]
```diff
@@ -390,7 +390,7 @@ protected Payload insertPayloadAndAttach(EReference reference, ID identifier, Pa
                     throw new IllegalArgumentException("Upper cardinality violated");
                 }
             } else {
-                if (container.get(reference.getName()) != null) {
+                if((container.get(reference.getName()) != null) ||  (!reference.isContainment() && !getNavigationResultAt(identifier, reference).isEmpty())) {
```
Comment (robertcsakany) [inline judo-runtime-core-dao-rdbms/src/main/java/hu/blackbelt/judo/runtime/core/dao/rdbms/RdbmsDAOImpl.java]: Space after IF!!!!!

**Review by robertcsakany — CHANGES_REQUESTED** [review]

File: judo-runtime-core-dao-rdbms/src/main/java/hu/blackbelt/judo/runtime/core/dao/rdbms/RdbmsDAOImpl.java (hunk 1 above)
Comment (gaborflorian) [reply judo-runtime-core-dao-rdbms/src/main/java/hu/blackbelt/judo/runtime/core/dao/rdbms/RdbmsDAOImpl.java]: [57f15d0](https://github.com/BlackBeltTechnology/judo-runtime-core/pull/204/commits/57f15d0e65e39a677fa99a79378b5c6bb4bc838b)

File: judo-runtime-core-dao-rdbms/src/main/java/hu/blackbelt/judo/runtime/core/dao/rdbms/RdbmsDAOImpl.java [hunk 2]
```diff
@@ -390,7 +390,7 @@ protected Payload insertPayloadAndAttach(EReference reference, ID identifier, Pa
                     throw new IllegalArgumentException("Upper cardinality violated");
                 }
             } else {
-                if (container.get(reference.getName()) != null) {
+                if ((container.get(reference.getName()) != null) ||  (!reference.isContainment() && !getNavigationResultAt(identifier, reference).isEmpty())) {
```
Comment (bencegelei) [inline judo-runtime-core-dao-rdbms/src/main/java/hu/blackbelt/judo/runtime/core/dao/rdbms/RdbmsDAOImpl.java:393]: Why does the reference being not containment (and the relation not empty) mean containment is already set?

Comment (bencegelei) [reply judo-runtime-core-dao-rdbms/src/main/java/hu/blackbelt/judo/runtime/core/dao/rdbms/RdbmsDAOImpl.java:393]: OK, so after some clarification: apparently reference - the mapping reference - is association and its mapped reference is the containment. Comment resolved

**Review by bencegelei — APPROVED** [review]

**Pull request merged**